src/machines/triviaMachine.ts

```diff
@@ -0,0 +1,252 @@
+import { assign, fromPromise, not, setup } from "xstate";
+import { RMCharacter } from "../common/types";
+import { RickCharacters } from "../services/RickApi";
+import { getRandomNumber } from "../common/constants";;
+
+const characterLogic = fromPromise(async () => {
+    const randomNumber = getRandomNumber();
+    const character = await RickCharacters.getCharacter(randomNumber)
+    return character;
+});
+
+
+const triviaMachine = setup({
+    types: {
+        events: {} as 
+        | { type: "user.play"} 
+        | { type: "user.close"} 
+        | { type: "user.reject"}
+        | { type: "user.accept"}
+        | { 
+            type: "user.selectAnswer",
+            answer: number
+        }
+        | { type: "user.nextQuestion"}
+        | { type: "user.toggleClue"}
+        | { type: "user.playAgain"},
+        context: {} as {
+            homePageCharacters: Array<RMCharacter>
+            hasLoaded: boolean
+            currentCharacter: RMCharacter | null
+            randomCharacters: Array<RMCharacter>
+            isClueOpened: boolean
+            points: number
+            question: number
+            lifes: number
+        }
+    },
```

Comment:
This is the part of the machine where we define the types for events and context. Is really useful to add them in order to handle autocomplete and get errors in case we are trying to update data that doesn't exist in the context or has the wrong value.